### DISCUSSION TITLE
Is there a way to add a folder to leptos watch ?

### DISCUSSION THREAD

**Discussion by Baptistemontan** [leptos-rs/leptos#1595] [Q&A]
[leptos_i18n](https://github.com/Baptistemontan/leptos_i18n) loads files in a folder called `locales` in the root of the project, is there a way to add this folder to cargo watch so when the locales change the app reload ?

**Comment by gbj** [comment]
Oh hm, interesting. Here's the logic for which files are watched:
https://github.com/leptos-rs/cargo-leptos/blob/ba4033f0dfdd7a95bd2254a0a73a6f7381e9a063/src/service/notify.rs#L18-L38

- `lib/bin` dependencies (Cargo.toml rust dependencies and your local src)
- `js_dir` 
- `assets` directory
- style and Tailwind-related files

I wonder if something hacky like `js-dir = "locales"` in your Cargo.toml cargo-leptos config would do it.

Maybe there should be a "watch additional files" config option here.

**Reply by Baptistemontan** [reply]
Interesting, when using `js-dir = "locales"` and watching it does output 
```
Notify watching folders  locales, public, src, *other folders*..
```
So `locales` appear in the watched folder, but modifying a file in the folder and saving it does not trigger a reload.

**Reply by Baptistemontan** [reply]
hoooo I get it, I added a  `whatever.js` file in the `locales` folder, and it does track it, so it tracks only javascript files, not json files

**Reply by Baptistemontan** [reply]
An option for additionnal files would be great, there would probably be other situations where this could be handy

**Reply by Baptistemontan** [reply]
yeah `js-dir` only watch javascript files (not even typescript files): https://github.com/leptos-rs/cargo-leptos/blob/ba4033f0dfdd7a95bd2254a0a73a6f7381e9a063/src/service/notify.rs#L99

**Comment by Baptistemontan** [comment]
I made a PR to add such feature: https://github.com/leptos-rs/cargo-leptos/pull/191
